lib/conn.c

```diff
@@ -33,24 +36,45 @@ bool conn_init(conn_t *conn, char *addr_str, int port)
     if(!pktbuf_init(&conn->pktbuf))
         return false;
 
-    conn->listen_fd = socket(AF_INET, SOCK_STREAM, 0);
-    if (conn->listen_fd < 0)
-        return false;
-
-    int optval = 1;
-    if (setsockopt(conn->listen_fd, SOL_SOCKET, SO_REUSEADDR, &optval,
-                   sizeof(optval)) < 0) {
-        warn("Set sockopt fail.\n");
-        goto fail;
-    }
-
-    struct sockaddr_in addr;
-    addr.sin_family = AF_INET;
-    addr.sin_addr.s_addr = inet_addr(addr_str);
-    addr.sin_port = htons(port);
-    if (bind(conn->listen_fd, (struct sockaddr *) &addr, sizeof(addr)) < 0) {
-        warn("Bind fail.\n");
-        goto fail;
+    struct in_addr addr_ip;
+    int is_inet_addr = inet_aton(addr_str, &addr_ip);
+    if (is_inet_addr == 1) {
+        if (!port) {
```

Comment:
Is there any reason to check on this specifically? Since `conn.c`  is not exported to the library user, we can assume the `port` is always given correctly here.

Reply:
> Is there any reason to check on this specifically? Since `conn.c` is not exported to the library user, we can assume the `port` is always given correctly here.

The port specified in this function is parsed from a user-provided string found [here](https://github.com/RinHizakura/mini-gdbstub/blob/415eaeb5107386c40c62755bdcc59b41ded5ff01/lib/gdbstub.c#L80), and this check verifies whether a port is missing, as in '127.0.0.1'.

Reply:
No strict restrictions say that port 0 is always encoded as invalid. Since you have checked the `port` parsed correctly before calling `conn_init()` and an incorrect port should fail at `bind()`, it could be possible to skip this check,